modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/ServiceTemplateManage.java

```diff
@@ -485,10 +473,76 @@ public String getOpenApiUrl(UUID id) {
         return openApiUrl;
     }
 
+    /**
+     * List service template history with query parameters.
+     *
+     * @param serviceTemplateId id of service template.
+     * @param requestType       type of service template request.
+     * @param changeStatus      status of service template request.
+     * @return list of service template history.
+     */
+    public List<ServiceTemplateHistoryVo> listServiceTemplateHistory(
+            UUID serviceTemplateId, ServiceTemplateRequestType requestType,
+            ServiceTemplateChangeStatus changeStatus) {
+        ServiceTemplateEntity existingTemplate =
+                getServiceTemplateDetails(serviceTemplateId, true, false);
+        List<ServiceTemplateHistoryEntity> historyList =
+                existingTemplate.getServiceTemplateHistory();
+        if (Objects.nonNull(requestType)) {
+            historyList = historyList.stream()
+                    .filter(history -> history.getRequestType() == requestType)
+                    .toList();
+        }
+        if (Objects.nonNull(changeStatus)) {
+            historyList = historyList.stream()
+                    .filter(history -> history.getStatus() == changeStatus)
+                    .toList();
+        }
+        return historyList.stream()
+                .sorted(Comparator.comparing(ServiceTemplateHistoryEntity::getCreateTime)
```

Comment:
this is good. can we also write this in the API doc - the history returned is sorted by the ascending order of the change requested time.